fondat/aws/bedrock/resources/sessions.py

```diff
@@ -0,0 +1,289 @@
+        """
+        async with runtime_client(self.config_runtime) as client:
+            await client.delete_session(sessionIdentifier=sessionIdentifier)
+
+    @operation(method="post", policies=lambda self: self.policies)
+    async def end(self, sessionIdentifier: str) -> Mapping[str, Any]:
+        """End session.
+
+        Args:
+            sessionIdentifier: Session identifier
+
+        Returns:
+            Session end details
+        """
+        async with runtime_client(self.config_runtime) as client:
+            return await client.end_session(sessionIdentifier=sessionIdentifier)
+
+    @operation(method="patch", policies=lambda self: self.policies)
+    async def update(
+        self,
+        sessionIdentifier: str,
+        *,
+        sessionMetadata: dict[str, str] | None = None,
+    ) -> Mapping[str, Any]:
+        """Update session metadata.
+
+        Args:
+            sessionIdentifier: Session identifier
+            sessionMetadata: Session metadata
+
+        Returns:
+            Updated session details
+        """
+        params = {"sessionIdentifier": sessionIdentifier}
+        if sessionMetadata is not None:
+            params["sessionMetadata"] = sessionMetadata
+        async with runtime_client(self.config_runtime) as client:
+            return await client.update_session(**params)
+
+    @property
+    def invocations(self) -> "InvocationsResource":
+        """Get invocations resource."""
+        return InvocationsResource(
+            self._agent_id, config_runtime=self.config_runtime, policies=self.policies
+        )
+
+
+@resource
+class InvocationsResource:
+    """Resource for managing session invocations."""
+
+    __slots__ = ("_agent_id", "config_runtime", "policies")
+
+    def __init__(
+        self,
+        agent_id: str,
+        *,
+        config_runtime: Config | None = None,
+        policies: Iterable[Policy] | None = None,
+    ):
+        self._agent_id = agent_id
+        self.config_runtime = config_runtime
+        self.policies = policies
+
+    @operation(method="post", policies=lambda self: self.policies)
+    async def create(
+        self,
+        sessionIdentifier: str,
+        invocationId: str,
+        *,
+        description: str | None = None,
+    ) -> Mapping[str, Any]:
+        """Create invocation.
+
+        Args:
+            sessionIdentifier: Session identifier
+            invocationId: Invocation identifier
+            description: Invocation description
+
+        Returns:
+            Invocation details
+        """
+        params = {
+            "sessionIdentifier": sessionIdentifier,
+            "invocationId": invocationId,
+        }
+        if description is not None:
+            params["description"] = description
+        async with runtime_client(self.config_runtime) as client:
+            return await client.create_invocation(**params)
+
+    @operation(method="post", policies=lambda self: self.policies)
```

Comment:
Wouldn't it be better to create a `InvocationResource` and just use a regular `PUT` as an operation ?